modules/compute/src/test/java/org/apache/ignite/internal/compute/queue/PriorityQueueExecutorTest.java

```diff
@@ -199,21 +196,142 @@ public void testQueueOverflow() {
         CountDownLatch latch2 = new CountDownLatch(1);
 
         //Submit task for executing
-        priorityQueueExecutor.submit(() -> {
+        submit(() -> {
             latch1.await();
             return 0;
         }, 1);
 
         //Submit task for executing, should be in queue
-        priorityQueueExecutor.submit(() -> {
+        submit(() -> {
             latch2.await();
             return 1;
         }, 1);
 
         //Submit task for execution should throw exception because queue is full.
         assertThat(
-                priorityQueueExecutor.submit(() -> null),
+                submit(() -> null),
                 willThrow(IgniteException.class, "Compute queue overflow")
         );
+
+        // Force the tasks to exit to minimize executor shutdown time
+        latch1.countDown();
+        latch2.countDown();
+    }
+
+    @Test
+    void taskCatchesInterruption() {
+        initExecutor(1);
+
+        CountDownLatch latch = new CountDownLatch(1);
+        QueueExecution<Object> execution = priorityQueueExecutor.submit(() -> {
+            while (true) {
+                try {
+                    latch.await();
+                } catch (InterruptedException e) {
+                    Thread.currentThread().interrupt();
```

Comment:
Can we submit one more job to the single-threaded executor after this one is canceled? I think `Thread.currentThread().interrupt()` makes future executions impossible.